### DISCUSSION TITLE
当有多个屏幕时，启动PRemoteM不能很好的记住关闭时所在的屏幕

### DISCUSSION THREAD

**Discussion by bgycn** [1Remote/1Remote#517] [Ideas]
**Describe the bug**
当有多个屏幕时，启动PRemoteM不能很好的记住关闭时所在的屏幕。特别是副屏幕的分辨率大于主屏幕，并且位于主屏幕上方（此时副屏幕的坐标为负数，猜测是导致问题的原因），每次重新启动PRemoteM则会自动跑到主屏幕，并且打开的远程桌面窗口也会跑到主屏幕，并且窗口大小会被压缩到主屏幕的分辨率。

**To Reproduce**
1.启动PRemoteM，把它拖到到分辨率更大的副屏幕，双击打开某个远程桌面，打开窗口也拖动到副屏幕；
2.退出PRemoteM；
3.再次启动PRemoteM，则它会自动跑到主屏幕，双击打开某个远程桌面则会把窗口大小压缩到不成比例。

**Expected behavior**
重新启动PRemoteM，应该记住之前所在的屏幕和位置，而不是每次都在主屏幕启动。

**Desktop (please complete the following information):**
 - OS: Windows 11
 - PRemoteM Version 0.7.2.8

**Comment by VShawn** [comment]
屏幕定位逻辑其实是如果当前没有tab窗口，那么当前鼠标在哪个屏幕，就会将新的远程桌面打开在哪个屏幕。

记住上次屏幕所在位置有一个问题，就是如果上次记忆在B屏幕，但当前已经有一个tab窗口在A屏幕了，那么新打开的远程桌面到底应该在B 屏幕还是A屏幕。如果在A屏幕，那么屏幕位置记忆功能就是没用的，如果在B屏幕，那么就会新创建一个tab窗口，这和tab窗口的设计又有矛盾。

以前是考虑过上述问题之后，最终才决定不再记忆上次关闭的位置，转而以当前鼠标焦点屏幕作为默认的显示屏幕。

**Comment by bgycn** [comment]
这样的话，会导致每次启动都是在主屏幕，而打开远程桌面，就会在主屏幕显示，然后因为主屏幕分辨率较小，窗口会被压缩变形，每次都要把PRemoteM移动到副屏重新设置窗口大小。或者我建议可以增加一个功能，用户可以配置为不使用tab窗口，而是每个远程远程桌面单独一个窗口，各自记住自己的位置，这样对于超大分辨率屏幕的用户来说比较友好，因为打开的远程桌面窗口通常只有屏幕的很小一部分，并且希望多个远程桌面并列排布，现在要实现这样就必须打开远程桌面之后，手动拖动到屏幕指定位置，还要调整窗口大小，很麻烦。

**Comment by VShawn** [comment]
> 这样的话，会导致每次启动都是在主屏幕

并不会这样，鼠标在副屏，就会在副屏显示。如果你不想挪动主窗口到副屏，那么当鼠标在副屏的时候就用 alt + m 启动器好了。

> 或者我建议可以增加一个功能，用户可以配置为不使用 tab 窗口，而是每个远程远程桌面单独一个窗口，各自记住自己的位置

以前考虑过这个方案，但一直没用空去仔细分析细节，比如说有的用户可能希望一部分服务器记住位置，一部分又自动用tab显示。
